### ISSUE TITLE
Error installing and starting docker service

### ISSUE THREAD

**Issue by pedroabi** [chef-cookbooks/docker#743]
### Cookbook version
2.9.5

### Chef-client version
Chef Development Kit Version: 0.15.16
chef-client version: 12.11.18
delivery version: master (444effdf9c81908795e88157f01cd667a6c43b5f)
berks version: 4.3.5
kitchen version: 1.10.0

### Platform Details
Vagrant: bento/ubuntu-14.04

### Scenario:
I am trying to install and start docker service.
Added on metadata.rb: 
depends 'docker', '~> 2.9.5'

Added on recipes/default.rb: 
docker_service 'default' do
  action [:create, :start]
end

### Steps to Reproduce:
Just run the recipes/default.rb:
docker_service 'default' do
  action [:create, :start]
end

### Expected Result:
?

### Actual Result:
==> default: [2016-08-24T16:47:26+00:00] INFO: Running queued delayed notifications before re-raising exception
==> default: 
==> default: Running handlers:
==> default: [2016-08-24T16:47:26+00:00] ERROR: Running exception handlers
==> default: Running handlers complete
==> default: 
==> default: [2016-08-24T16:47:26+00:00] ERROR: Exception handlers complete
==> default: Chef Client failed. 0 resources updated in 05 seconds
==> default: [2016-08-24T16:47:26+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> default: [2016-08-24T16:47:26+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
==> default: [2016-08-24T16:47:26+00:00] ERROR: docker_service[default] (zx-airflow::default line 9) had an error: NoMethodError: docker_service_manager_upstart[default] (/var/chef/cache/cookbooks/docker/libraries/docker_service.rb line 74) had an error: NoMethodError: service[docker] (/var/chef/cache/cookbooks/docker/libraries/docker_service_manager_upstart.rb line 36) had an error: NoMethodError: undefined method `[]' for nil:NilClass
==> default: [2016-08-24T16:47:27+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.

**Comment by etilabs** [comment]
I had a similar problem, and have the same versions of Chefdk and Knife:

```Running handlers:
[2016-08-25T15:20:02+00:00] ERROR: Running exception handlers
Running handlers complete
[2016-08-25T15:20:02+00:00] ERROR: Exception handlers complete
Chef Client failed. 9 resources updated in 16 seconds
[2016-08-25T15:20:02+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
[2016-08-25T15:20:02+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2016-08-25T15:20:02+00:00] ERROR: docker_service[default] (blockchain::docker-host line 107) had an error: Errno::ENOENT: docker_service_manager_upstart[default] (/var/chef/cache/cookbooks/docker/libraries/docker_service.rb line 74) had an error: Errno::ENOENT: No such file or directory - /usr/bin/docker
[2016-08-25T15:20:02+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)```

The root of both of these may be that Docker is just not really being installed from that package.  I had this happen earlier today:

```
# docker ps
The program 'docker' is currently not installed. You can install it by typing:
apt-get install docker
# apt-get install -y docker
Reading package lists... Done
Building dependency tree       
Reading state information... Done
docker is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 35 not upgraded.
```

**Comment by tduffield** [comment]
What version of the docker-engine got installed?

**Comment by tduffield** [comment]
I got this error as well. We solved it by downgrading docker-engine. We pinned to 1.10.3, you're results may vary.

**Comment by ritterwolf** [comment]
I suspect I know the cause of this. For new versions of docker-engine, the daemon executable is `/usr/bin/dockerd`; but the Upstart job is running `/usr/bin/docker`.

I'll do some testing to confirm it.

**Comment by ritterwolf** [comment]
PR #741 fixed this for me.

**Comment by etilabs** [comment]
@ritterwolf I had noticed the same thing in my testing.  Doesn't that signify that it could be a problem with the `apt` package rather than anything in the Chef cookbook?

**Comment by chasebolt** [comment]
I'll get PR #741 tested and merged in today

**Comment by chasebolt** [comment]
resolves in v2.9.6
